### PR TITLE
feat(cart): create cart service integration

### PR DESCRIPTION
**Pull request by Sinucid**
Integration of edit cart component with services. 

- `create` method of cart adapter is resolving default price mode and currency only if qualifier is not provided
- `ContentPageMetaResolver` resolves page's meta via route type instead of route path now
-  form element emits **oryx.submit** event instead of host element in `FormController` now

closes: [HRZ-90752](https://spryker.atlassian.net/browse/HRZ-90752)

[HRZ-90752]: https://spryker.atlassian.net/browse/HRZ-90752?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/cart/edit/edit.component.ts [hunk 1]
```diff
@@ -1,39 +1,48 @@
...
-import { computed, signal } from '@spryker-oryx/utilities';
+import { AlertType } from '@spryker-oryx/ui';
+import { I18nMixin, computed, signal } from '@spryker-oryx/utilities';
 import { LitElement, TemplateResult, html } from 'lit';
-import { CartEditComponentOptions } from './edit.model';
+import { state } from 'lit/decorators.js';
 
-@defaultOptions({ isDefault: true })
 export class CartEditComponent extends CartComponentMixin(
-  FormMixin(ContentMixin<CartEditComponentOptions>(LitElement))
```
Comment (tobi-or-not-tobi) [inline libs/domain/cart/edit/edit.component.ts:23]: Can we drop the component model as well? Either remove or deprecate, depending on when we introduced it

File: libs/domain/cart/edit/edit.component.ts [hunk 2]
```diff
@@ -111,30 +110,38 @@ export class CartEditComponent extends CartComponentMixin(
     const fields = this.getFields();
     const values = this.$values();
 
-    return html`<form @submit=${this.onSubmit}>
+    return html`<form @oryx.submit=${this.onSubmit}>
```
Comment (supproduction) [inline libs/domain/cart/edit/edit.component.ts:113]: looks like it's BC

File: libs/domain/cart/edit/edit.component.ts [hunk 3]
```diff
@@ -111,30 +110,38 @@ export class CartEditComponent extends CartComponentMixin(
...
+          //TODO: use translation string when issue with resolving of translated result
+          //outside components render is fixed
+          // content: this.i18n('carts.create.cart-<name>-created', { name: cart.name }) as string
+          content: `Cart: ${cart.name} was created`,
+        });
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        this.routerService.navigate(this.$redirectLink()!);
+      },
+      complete: () => {
+        this.isLoading = false;
```
Comment (supproduction) [inline libs/domain/cart/edit/edit.component.ts:143]: Do we cover error?

File: libs/domain/cart/services/src/adapter/default-cart.adapter.ts [hunk 4]
```diff
@@ -203,27 +203,32 @@ export class DefaultCartAdapter implements CartAdapter {
...
 
   create(qualifier?: CreateCartQualifier): Observable<Cart> {
-    return combineLatest([
-      this.store.get(),
-      this.currency.get(),
-      this.priceMode.get(),
-    ]).pipe(
+    const request = (attributes: CreateCartQualifier) =>
+      this.http.post<ApiCartModel.Response>(
+        `${this.SCOS_BASE_URL}/${ApiCartModel.UrlParts.Carts}`,
```
Comment (supproduction) [inline libs/domain/cart/services/src/adapter/default-cart.adapter.ts:208]: I wonder can we start using endpoint service here cc @dunqan

File: libs/platform/experience/src/services/resolvers/content-page-meta.resolver.ts [hunk 5]
```diff
@@ -68,19 +73,32 @@ export class ContentPageMetaResolver implements PageMetaResolver {
...
-    })?.meta;
+  /** @deprecated since 1.4. use RouteWithParams as parameter  */
+  protected getData(route: string): ExperienceComponent['meta'] | undefined;
+  protected getData(
+    route: RouteWithParams
+  ): ExperienceComponent['meta'] | undefined;
+  protected getData(
+    route: RouteWithParams | string
+  ): ExperienceComponent['meta'] | undefined {
+    if (typeof route === 'string') {
```
Comment (supproduction) [inline libs/platform/experience/src/services/resolvers/content-page-meta.resolver.ts]: use featureVestion here and typecasting

**Review by supproduction — CHANGES_REQUESTED** [review]

File: libs/domain/cart/edit/edit.component.ts (hunk 2 above)
Comment (Sinucid) [reply libs/domain/cart/edit/edit.component.ts:113]: This component added as a part of this feature, have not yet released

File: libs/domain/cart/edit/edit.component.ts (hunk 3 above)
Comment (Sinucid) [reply libs/domain/cart/edit/edit.component.ts:143]: will be handled by our global error handler (show the notification)

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: libs/domain/cart/edit/edit.component.ts (hunk 1 above)
Comment (Sinucid) [reply libs/domain/cart/edit/edit.component.ts:23]: Do you meant the

File: libs/domain/cart/services/src/adapter/default-cart.adapter.ts (hunk 4 above)
Comment (dunqan) [reply libs/domain/cart/services/src/adapter/default-cart.adapter.ts:208]: we don't have endpoints services, but we have generic handling of includes

**Review by supproduction — APPROVED** [review]

**Review by tobi-or-not-tobi — APPROVED** [review]

**Pull request merged**